app/src/main/java/com/bebetterprogrammer/tictactoe/activities/GamePlayActivity.kt

```diff
@@ -73,40 +72,62 @@ class GamePlayActivity : AppCompatActivity() {
             LayoutInflater.from(v.context).inflate(R.layout.result_dialog, viewGroup, false)
         builder.setView(dialogView)
         val alertDialog: AlertDialog = builder.create()
-        dialogView.result.text = "Yeppii.. $playerName Won!"
+        if (!istie) {
+            dialogView.resultTrophy.setImageResource(R.drawable.ic_trophy_won)
+            dialogView.result.text = "Yeppii.. $playerName Won!"
+        } else {
+            dialogView.resultTrophy.setImageResource(R.drawable.ic_trophy_tie)
+            dialogView.result.text = playerName
+        }
+        alertDialog.show()
 
-        dialogView.btnRematch.setOnClickListener { alertDialog.dismiss() }
+        dialogView.btnRematch.setOnClickListener {
+            reset()
+            alertDialog.dismiss()
+        }
 
         dialogView.btnQuit.setOnClickListener {
             val intent = Intent(this, HomePageActivity::class.java)
             startActivity(intent)
         }
-        alertDialog.show()
+    }
+
+    private fun reset() {
+        for (i in 0..8) {
+            gameState[i] = 2
+        }
+        var q = listOf<ImageView>(btn1, btn2, btn3, btn4, btn5, btn6, btn7, btn8, btn9)
+        for (i: ImageView in q) {
+            i.setImageResource(0)
+        }
+        won = 0
+        iswon = false
+        istie = false
+        turn = pl
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_gameplay)
-
         val versionName = BuildConfig.VERSION_NAME
         appBottomLine.text = "Designed @ bebetterprogrammer.com | v$versionName"
-
-        val intent = intent
-        P1 = intent.getStringExtra("Player1")
-        P2 = intent.getStringExtra("Player2")
+        P1 = intent.getStringExtra("Player1") ?: "Player1"
+        P2 = intent.getStringExtra("Player2") ?: "Player2"
         val Player = intent.getIntExtra("Player", 0)
         Player1.text = P1
         Player2.text = P2
         if (Player == 0) {
             first = 0
             turn = 0
+            pl = 0
         } else if (Player == 1) {
             turn = 1
             first = 1
+            pl = 1
         }
         quit.setOnClickListener {
-            val intent = Intent(this, HomePageActivity::class.java)
-            startActivity(intent)
+            val intentx = Intent(this, HomePageActivity::class.java)
```

Comment:
Please provide the valid name to a variable. Either `intent` or `homePageActivityIntent`.